rfc/rfc022-discovery-service.md

```diff
@@ -0,0 +1,252 @@
+Users need to know which (remote) systems offer a particular service and systems need to know how to connect to those systems.
+
+This RFC defines a protocol for Service Discovery using Verifiable Presentations using a client-server model.
+The advantage of using Verifiable Presentations is that the client needs little trust in the server:
+clients can verify the authenticity and integrity of the Verifiable Presentation and Verifiable Credential.
+
+## 2. Terminology
+
+* **Discovery Service**: collection of Verifiable Presentations that allows discovery of systems/parties that offer a particular service.
+* **Discovery Definition**: document describing the Discovery Service.
+* **Server**: system hosting the discovery service and accepting new registrations.
+* **Client**: system registering presentations on the list and reading it to discover other systems/parties.
+
+Other terms are as defined in the following specifications: "JSON Web Token (JWT)" [JWT], Verifiable Credentials Data Model 1.1 [VC-DATA-MODEL], Verifiable Credentials Presentation Exchange [PE], Decentralized Identifiers 1.0 [DID].
+
+## 3. Service endpoint
+
+The service endpoint is an HTTP endpoint exposed by the server which hosts a list of Verifiable Presentations.
+Clients can query the list or submit a new Verifiable Presentation to it.
+All presentations MUST conform to the Presentation Definition associated with the discovery service (see Service Definition).
+
+Presentations MUST be encoded in JWT format as string ([JWT]).
+
+### 3.1 Registration
+
+Clients can publish a Verifiable Presentation to the list by sending it in an HTTP POST to the service endpoint.
+The HTTP request body MUST be a Verifiable Presentation (in JWT format). The content type of the request MUST be `application/json`.
+
+The server MUST validate the Verifiable Presentation as specified in section 4. If the validation fails, it MUST return a 400 Bad Request response.
+If the validation succeeds, the Verifiable Presentation MUST be added to the list and the server MUST return a 201 Created response.
+
+A credential subject (identified by `credentialSubject.id`) MUST NOT appear more than once on the list,
+so a new registration MUST replace the previous one from the same credential subject.
+
+An example posting a Verifiable Presentation in JWT format to the list:
+
+``http request
+POST /list HTTP/1.1
+Content-Type: application/json
+
+"eyCAFE.etc.etc"
+``
+
+The Verifiable Presentation MUST NOT be valid longer than the Verifiable Credentials it contains. 
+
+### 3.2 Reading
+
+Clients MUST read the list by sending an HTTP GET to the server's service endpoint.
+The server MUST return a 200 OK response with a JSON object containing:
+- `entries` REQUIRED. MUST contain a JSON array with zero or more presentations.
+- `tag` REQUIRED. MUST be a JSON string referring to the last entry. The tag is an opaque value; the client SHOULD NOT derive any meaning from it.
```

Comment:
> referring to the last entry.

It refers to the moment in time of the last entry (in the ref impl). A description like: "used by the server to provide delta's" might be better suited